### PR TITLE
cpu/lpc2387: clean up the platform

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description
For historic reasons, clock setup for lpc2387 was done in the board folder. This lead to code duplication between the two lpc2387 boards. The setup routines are exactly the same, `avsextrem` added two safety guards, I decided to keep those (even though `msba2` seemed to be doing fine without all those years).

There should be no code changes, code was only moved around.
Compile-tested only as I don't have the hardware.

### Testing procedure

Check that `msba2` and `avsextrem` still work.

### Issues/PRs references
`avsextrem` was forgotten in #11367, so this should also fix stdio.

### CODE REVIEW

**Comment by maribu** [comment]
Cool.l! I intended to do just this as well, but didn't found time. I'll test on Monday on the MSB-A2. I think no one has the avsextrem, but I'm not sure

File: cpu/lpc2387/cpu.c [hunk 1]
```diff
@@ -110,4 +113,22 @@ void arm_reset(void)
...
+ * @brief Initialize the CPU, set IRQ priorities, clocks
+ */
+void cpu_init(void)
+{
+    /* configure CPU clock */
+    cpu_init_clks();
+
+    /* set up GPIOs */
+    gpio_init_ports();
+
```
Comment (maribu) [inline cpu/lpc2387/cpu.c]: I think `board_init()` should be best called here, as it sets up UART0 needed for stdio.

Comment (benpicco) [reply cpu/lpc2387/cpu.c]: Ok.

**Comment by maribu** [comment]
Please squash. I'll do one final test then before merging. Thanks for this PR, this was a big step at bringing the boot process of the lpc2387 back up to standard :-)

**Comment by benpicco** [comment]
Squashed. 
Thank you for testing!

**Comment by maribu** [comment]
```patch

**Comment by benpicco** [comment]
Oops, sorry - fixed.

**Pull request merged**